DependencyInjection/FOSRestExtension.php

```diff
@@ -160,9 +160,11 @@ public function load(array $configs, ContainerBuilder $container)
             $container->setParameter($this->getAlias().'.decoders', $config['body_listener']['decoders']);
 
             $arrayNormalizer = $config['body_listener']['array_normalizer'];
-            if (null !== $arrayNormalizer) {
-                $container->getDefinition($this->getAlias().'.body_listener')
-                    ->addMethodCall('setArrayNormalizer', array(new Reference($arrayNormalizer)));
+
+            if (null !== $arrayNormalizer['service']) {
+                $bodyListener = $container->getDefinition($this->getAlias().'.body_listener');
+                $bodyListener->addArgument(new Reference($arrayNormalizer['service']));
+                $bodyListener->addArgument($arrayNormalizer['forms']);
```

Comment:
Thanks for the suggestion, but `replaceArgument()` will throw an exception if the argument was not added (it only replaces it).